scipy/sparse/linalg/_eigen/_svds.py

```diff
@@ -209,48 +211,238 @@ def svds(A, k=6, ncv=None, tol=0, which='LM', v0=None,
-    >>> np.allclose(A3, A.toarray())
+    >>> u5, s5, vT5 = svds(A, k=5)
+    >>> A5 = u5 @ np.diag(s5) @ vT5
+    >>> np.allclose(A5, A)
     True
 
-    The singular values match the expected singular values, and the singular
-    vectors are as expected up to a difference in sign.
+    The singular values match the expected singular values.
 
-    >>> np.allclose(s3, s)
+    >>> np.allclose(s5, s)
     True
-    >>> np.allclose(np.abs(u3), np.abs(u.toarray()))
-    True
-    >>> np.allclose(np.abs(vT3), np.abs(vT.toarray()))
+
+    Since the singular values are not close to each other in this example,
+    every singular vector matchs as expected up to a difference in sign.
+
+    >>> (np.allclose(np.abs(u5), np.abs(u)) and
+    ...  np.allclose(np.abs(vT5), np.abs(vT)))
     True
 
     The singular vectors are also orthogonal.
 
-    >>> np.allclose(u3.T @ u3, np.eye(5))
+    >>> (np.allclose(u5.T @ u5, np.eye(5)) and
+    ...  np.allclose(vT5 @ vT5.T, np.eye(5)))
+    True
+
+    If there are (nearly) multiple singular values, the corresponding
+    individual singular vectors may be unstable, but the whole invariant
+    subspace containing all such singular vectors is computed accurately
+    as can be measured by angles between subspaces via 'subspace_angles'.
+
+    >>> from scipy.linalg import subspace_angles as s_a
+    >>> rng = np.random.default_rng()
+    >>> s = [1, 1 + 1e-6]  # non-zero singular values
+    >>> u, _ = np.linalg.qr(rng.standard_normal((99, 2)))
+    >>> v, _ = np.linalg.qr(rng.standard_normal((99, 2)))
+    >>> vT = v.T
+    >>> A = u @ np.diag(s) @ vT
+    >>> A = A.astype(np.float32)
+    >>> u2, s2, vT2 = svds(A, k=2)
+    >>> np.allclose(s2, s)
+    True
+
+    The angles between the individual exact and computed singular vectors
+    are not so small.
+
+    >>> s_a(u2[:, :1], u[:, :1]) + s_a(u2[:, 1:], u[:, 1:]) > 1e-3
+    True
+
+    >>> (s_a(vT2[:1, :].T, vT[:1, :].T) +
+    ...  s_a(vT2[1:, :].T, vT[1:, :].T)) > 1e-3
+    True
+
+    As opposed to the angles between the 2-dimentional invariant subspaces
+    that these vectors span, which are small for rights singular vectors
+
+    >>> s_a(u2, u).sum() < 1e-6
+    True
+
+    as well as for left singular vectors.
+
+    >>> s_a(vT2.T, vT.T).sum() < 1e-6
+    True
+
+    The next example follows that of 'sklearn.decomposition.TruncatedSVD'.
+
+    >>> rng = np.random.RandomState(0)
+    >>> X_dense = rng.random(size=(100, 100))
+    >>> X_dense[:, 2 * np.arange(50)] = 0
+    >>> X = csr_matrix(X_dense)
+    >>> _, singular_values, _ = svds(X, k=5)
+    >>> print(singular_values)
+    [ 4.3293...  4.4491...  4.5420...  4.5987... 35.2410...]
+
+    The function can be called without the transpose of the input matrix
+    ever explicitely constructed.
+
+    >>> from scipy.linalg import svd
+    >>> from scipy.sparse import rand
+    >>> from scipy.sparse.linalg import aslinearoperator
+    >>> rng = np.random.RandomState(0)
```

Comment:
same here--legacy I think

Reply:
This example just follows verbatim 'sklearn.decomposition.TruncatedSVD'